### PR TITLE
ci/test-docs-generation: Attempt to use version-sets

### PR DESCRIPTION
**Pull request by abhinav**
Addresses the TODO and uses version-set generated by get-job-matrix
to decide which version of our various platforms should be used.

### CODE REVIEW

**Comment by abhinav** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12039** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/12039" target="_blank">Graphite</a> 
    * **PR #12048** <a href="https://app.graphite.dev/github/pr/pulumi/pulumi/12048" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/pulumi/pulumi/12048?utm_source=stack-comment).

**Comment by abhinav** [comment]
This appears to have worked.

Version set:
image

Go:
image

Node:
image

Python:
image

.NET:
image

**Review by AaronFriel — APPROVED** [review]

**Review by RobbieMcKinstry — APPROVED** [review]
One comment. Otherwise LGTM

File: .github/workflows/ci-test-docs-generation.yml [hunk 1]
```diff
@@ -24,8 +24,35 @@ permissions:
...
+          )
+          echo "::endgroup::"
+
+          echo "::group::Version set"
+          echo "$VERSION_SET" | yq -P '.'
+          echo "::endgroup::"
+
+          echo "::group::Set outputs"
+          ./.github/scripts/set-output version-set "${VERSION_SET}"
+          echo "::endgroup::"
```
[the remark this answers is not in the document]
Comment (abhinav) [reply .github/workflows/ci-test-docs-generation.yml:49]: They've deprecated echo-based set-output, but these commands are still supported.

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines

**Pull request merged**